odoo/addons/base/models/ir_mail_server.py

```diff
@@ -83,6 +92,24 @@ def extract_rfc2822_addresses(text):
     return valid_addresses
 
 
+def _verify_check_hostname_callback(cnx, x509, err_no, err_depth, return_code, *, hostname):
+    """Callback used for pyOpenSSL.verify_mode, by default it only check
+       :param:`err_no`, we enrich it to also verify that the SMTP server
+       :param:`hostname` matches the :param:`x509`'s Common Name (CN) or
+       Subject Alternative Name (SAN)."""
+    if err_no:
+        return False
+
+    if err_depth == 0:  # leaf certificate
+        peercert = {
+            "subject": ((("commonName", x509.get_subject().CN),),),
+            "subjectAltName": get_subj_alt_name(x509),
+        }
```

Comment:
can't we get this from the socket instead ? (with `getpeercert`)

Reply:
I guess you are talking about [getpeercert](https://docs.python.org/3/library/ssl.html#ssl.SSLSocket.getpeercert) from the stdlib. Here we are using pyopenssl and it seems that there is no such function https://www.pyopenssl.org/en/latest/search.html?q=getpeercert